src/schematools/ckan/_convert.py

```diff
@@ -0,0 +1,76 @@
+from collections import defaultdict
+from time import strftime
+from typing import Any, Optional
+
+from ..types import DatasetSchema
+
+
+def from_dataset(ds: DatasetSchema) -> dict[str, Any]:
+    """Convert a dataset to the CKAN format variant used by data.overheid.nl.
+
+    The output, as JSON, can be used in a CKAN create_package request.
+
+    Documentation for the format can be pieced together from:
+    * https://docs.ckan.org/en/2.9/api/index.html
+    * https://github.com/dataoverheid/readthedocs-ckanext-dcatdonl/blob/master/source/usage-creationandupdating.rst
+    * https://waardelijsten.dcat-ap-donl.nl/
+    """
+
+    # XXX CKAN allows a dataset to have multiple resources associated with it,
+    # which we could use to represent the tables, but data.overheid.nl doesn't
+    # seem to support that (or it's broken).
+
+    # XXX This is not always correct, but the logic to construct paths lives in DSO-API.
+    # Only the Haal Centraal remotes have custom paths, though.
+    # Maybe get rid of the custom paths?
+    url = "https://api.data.amsterdam.nl/v1/" + ds.id
+
+    has_geo = any(table.has_geometry_fields for table in ds.tables)
+    theme = THEME_RUIMTE if has_geo else THEME_BESTUUR
+
+    return {
+        "authority": "http://standaarden.overheid.nl/owms/terms/Amsterdam",
+        "contact_point_name": "Gemeente Amsterdam",
+        "contact_point_email": "datapunt@amsterdam.nl",
+        "contact_point_website": "https://data.amsterdam.nl",
+        "format": JSON,
+        "identifier": url,  # This seems to work.
+        "language": [NLD],
+        "license_id": LICENSES[ds.license],
+        "metadata_language": NLD,
+        "mimetype": "application/json",
+        "modified": strftime("%Y-%m-%dT%H:%M:%S"),
+        "name": ds.title,
+        "notes": ds.description or ds.title,  # Must not be empty.
+        "owner_org": "gemeente-amsterdam",
+        "publisher": "http://standaarden.overheid.nl/owms/terms/Amsterdam",
+        "theme": [theme],
+        "title": ds.title,
+        # XXX This is not always correct, but the logic to construct paths lives in DSO-API.
+        # Only the Haal Centraal remotes have custom paths, though.
+        # Maybe get rid of the custom paths?
+        "url": "https://api.data.amsterdam.nl/v1/" + ds.id,
+    }
+
+
+# JSON is our primary file type. Not sure if we can submit multiple types for a dataset.
+# Keep this consistent with the mimetype.
+# From https://waardelijsten.dcat-ap-donl.nl/mdr_filetype_nal.json
+JSON = "http://publications.europa.eu/resource/authority/file-type/JSON"
+
+# From https://waardelijsten.dcat-ap-donl.nl/overheid_license.json
+LICENSES: dict[Optional[str], str] = defaultdict(
+    lambda: "http://standaarden.overheid.nl/owms/terms/licentieonbekend",
```

Comment:
`LICENSES.get("...", UNKNOWN_LICENSE)` would be equally clear to understand.

Reply:
Done.